src/main/java/edu/hillel/homework/lesson17/petrol_station/PetrolStation.java

```diff
@@ -0,0 +1,75 @@
+package edu.hillel.homework.lesson17.petrol_station;
+
+import com.google.common.util.concurrent.AtomicDouble;
+import edu.hillel.homework.lesson17.petrol_station.car.Car;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.concurrent.*;
+
+public class PetrolStation {
+
+    private static final int MAXIMUM_THREADS_ALLOWED = 3;
+    private final AtomicDouble amount;
+    private final ExecutorService executor;
+    private final CountDownLatch latch;
```

Comment:
Nice to see that you use extra implementation of synchronizator
But!
Here it's not necessary. By the current logic, autos should wait till the last one starts to be refueling. But in real life, autos should make a refuel independently from each other